### PR TITLE
Run clang-format on the repo, add a github action to check formatting

### PR DESCRIPTION
**Pull request by SLieve**
* Added a few clang-format directives to maintain ordering of includes:
  ```
  <angle-bracket includes> at the top

  <roaring/anything> below that

  "quoted includes" below that
  ```
* Formatted the repo with `tools/clang-format.sh`
* Marked the formatting commit as to-be-ignored by git blame, this should work with github too.
* Reordered some includes where there was still breakage, added comments to prevent futher reordering.
* Added a github action to check format against clang-format.
  * Note: This adds a github action dependency on https://github.com/DoozyX/clang-format-lint-action

### CODE REVIEW

**Comment by lemire** [comment]
Can you add instructions related to clang-cl under: `.github/PULL_REQUEST_TEMPLATE.md` and maybe somewhere as part of the README.md, or in an CONTRIBUTING.md file.

I don't object to your change, but I don't want to have to explain to people contributed code that they need to use clang-cl. This needs to be documented. Otherwise it creates a lot of bureaucratic exchanges.

Be mindful that not everyone uses clang-cl, so that's not entirely contributor friendly. (I personally do not object, to be clear.)

**Comment by SLieve** [comment]
Done. Note that the requirements are actually even a bit more strict: run a specific version of clang-format (currently stable version 17). This avoids inconsistent formatting. If you have any suggestion on making this smoother, I'm happy to take them onboard.

**Comment by lemire** [comment]
@Dr-Emann Can you review?

**Comment by Dr-Emann** [comment]
Might be worth a dockerfile with clang format installed, and a script that tries clang-format (checking the version), clang-format-17 and if neither work, using a docker container with the code mounted as a volume, to make it easier to format locally.

**Comment by lemire** [comment]
@Dr-Emann I code something up with Docker: https://github.com/RoaringBitmap/CRoaring/pull/580

Requiring users to have docker is a bit bad, but at least it will be consistent once they have it. Chasing the right clang-format version is terrible.

**Comment by SLieve** [comment]
Nice, yes using a bundled docker container sounds much nicer.

**Review by lemire — APPROVED** [review]
We should be able to merge this.

File: README.md [hunk 1]
```diff
@@ -882,6 +882,10 @@ Justin Whear wrote a Zig wrapper available at https://github.com/jwhear/roaring-
 
 https://groups.google.com/forum/#!forum/roaring-bitmaps
 
+# Contributing
+
+When contributing a change to the project, please run `clang-format` using Clang Format version 17 after making any changes. If Clang Format 17 is installed, this can also be accomplished by running the formatting script in `tools/clang-format.sh`. A github action runs on all PRs to ensure formatting is consistent with this.
```
Comment (Dr-Emann) [inline README.md:887]: This could be updated now that you don't need to have clang-format 17 installed locally if you have docker

**Review by Dr-Emann — APPROVED** [review]
Personally prefer a longer line length, but happy to have automated formatting, like they say about gofmt:

Gofmt's style is no one's favorite, yet gofmt is everyone's favorite.

**Pull request merged**